src/main/scala/firrtl/passes/Inline.scala

```diff
@@ -187,7 +187,10 @@ class InlineInstances extends Transform with DependencyAPIMigration with Registe
       renameMap:     RenameMap
     )(s:             Statement
     ): Statement = {
-      def onName(ofModuleOpt: Option[String])(name: String) = {
+      def onName(ofModuleOpt: Option[String])(name: String): String = {
+        // Empty names are allowed for backwards compatibility reasons and
+        // indicate that the entity has essentially no name and thus cannot be prefixed.
+        if (name.isEmpty) return name
```

Comment:
I did not know that it was implemented via exceptions. I thought it would just be a simple control flow operation like an if/else

Reply:
Seems like it isn't always implemented as an exception. Maybe we should add `-Xlint:nonlocal-return` to our `scalac` flags (only supported in Scala 2.13 + apparently)

Reply:
Save those early returns for C++... 😛

Reply:
It looks like the Scala standard library uses `return` sometimes so maybe it's fine. It's probably slow when the return is nested in anonymous functions but fine otherwise.